src/Microsoft.Framework.Runtime/DependencyManagement/LockFile.cs

```diff
@@ -15,13 +16,34 @@ public class LockFile
         public IList<LockFileLibrary> Libraries { get; set; } = new List<LockFileLibrary>();
         public IList<LockFileTarget> Targets { get; set; } = new List<LockFileTarget>();
 
+        public IEnumerable<ICompilationMessage> GetDiagnostics(Project project)
+        {
+            string message;
+            if (!IsValidForProject(project, out message))
+            {
+                yield return new FileFormatMessage(
+                    $"{message}. Please run \"dnu restore\" to generate a new lock file.",
+                    Path.Combine(project.ProjectDirectory, LockFileReader.LockFileName),
+                    CompilationMessageSeverity.Error);
+            }
+        }
+
         public bool IsValidForProject(Project project)
         {
+            string message;
+            return IsValidForProject(project, out message);
```

Comment:
I'd like to share my concerns here. It allocate a string but thrown away. It is understandable that the overload helps to reuse code, but from the perspective of performance it's not ideal.